### PR TITLE
Cleanup after #471

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Just some cleanup, separated from the big PR for ease of review.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

File: sdk/src/client/secret/ledger_nano.rs [hunk 1]
```diff
@@ -400,24 +393,19 @@ impl SecretManagerConfig for LedgerSecretManager {
...
-                // not fine, return
-                return true;
-            }
-        }
+    let TransactionEssence::Regular(essence) = &prepared_transaction.essence;
+
+    if essence
+        .outputs()
+        .iter()
+        .any(|output| !matches!(output, Output::Basic(o) if o.simple_deposit_address().is_some()))
```
Comment (Alex6323) [inline sdk/src/client/secret/ledger_nano.rs]: ah ... you prefer this version ... any specific reason why?

File: sdk/src/client/secret/mod.rs [hunk 2]
```diff
@@ -369,6 +369,16 @@ impl SecretManage for SecretManager {
...
 
+pub trait DowncastSecretManager: SecretManage {
+    fn downcast<'a, T: 'static + SecretManage>(&'a self) -> Option<&'a T>;
+}
+
+impl<S: 'static + SecretManage + Send + Sync> DowncastSecretManager for S {
+    fn downcast<'a, T: 'static + SecretManage>(&'a self) -> Option<&'a T> {
+        (self as &(dyn std::any::Any + Send + Sync)).downcast_ref::<T>()
+    }
+}
```
Comment (Alex6323) [inline sdk/src/client/secret/mod.rs]: that's fancy!!

File: sdk/src/client/secret/ledger_nano.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/client/secret/ledger_nano.rs]: No, I just glanced at your comment before writing it so I happened to do the opposite. They should be functionally identical.

File: sdk/src/client/secret/mod.rs (hunk 2 above)
Comment (DaughterOfMars) [reply sdk/src/client/secret/mod.rs]: Traits 😎

File: sdk/src/client/secret/ledger_nano.rs (hunk 1 above)
Comment (Alex6323) [reply sdk/src/client/secret/ledger_nano.rs]: yeah, they are. Had this one, too, but then switched to `.all`.

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**